File: leetcode/python/557.反转字符串的单词III.py
class Solution:
    def reverseWords(self, s: str) -> str:
        ss = s.split(' ')
        ret = []
        for string in ss:
            ret.append(self.reverseStr(string))
        return ' '.join(ret)

    def reverseStr(self, s: str) -> str:
        ret = ""
        for c in s:
            ret = c + ret
        return ret

File: leetcode/golang/557.反转字符串的单词III.go
import "strings"
func reverseWords(s string) string {
    ss := strings.Split(s, string(' '));
    var ret []string;
    for _, chs := range ss {
        tmp := reverse([]rune(chs));
        ret = append(ret, string(tmp));
    }
    return strings.Join(ret, string(' '));
};

func reverse(s []rune) []rune {
    var ret []rune;
    i := len(s) - 1;
    for {
        if i < 0 {
            break;
        }
        ret = append(ret, s[i]);
        i -= 1;
    }
    return ret;
}